infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -411,11 +438,57 @@ private DetectorTask detectFailure(PollReport pollReport, Layout layout) {
                 throw FailureDetectorException.layoutMismatch(clusterState, layout);
             }
 
+            DecisionMakerAgent decisionMakerAgent = new DecisionMakerAgent(clusterState, advisor);
+            Optional<String> maybeDecisionMaker = decisionMakerAgent.findDecisionMaker();
+
+            if (!maybeDecisionMaker.isPresent()) {
+                return DetectorTask.NOT_COMPLETED;
+            }
+
+            String decisionMaker = maybeDecisionMaker.get();
+
+            Optional<NodeRankByPartitionAttributes> maybeFailedNodeByPartitionAttr = fsAdvisor
+                    .findFailedNodeByPartitionAttributes(clusterState);
+
+            if (maybeFailedNodeByPartitionAttr.isPresent()) {
+                NodeRankByPartitionAttributes failedNode = maybeFailedNodeByPartitionAttr.get();
+
+                if (decisionMaker.equals(failedNode.getEndpoint())) {
+                    log.error("Decision maker and failed node are the same node: {}", decisionMakerAgent);
+                    return DetectorTask.NOT_COMPLETED;
+                }
+
+                Set<String> failedNodes = new HashSet<>();
+                failedNodes.add(failedNode.getEndpoint());
+                return detectFailure(layout, failedNodes, pollReport).join();
+            }
```

Comment:
What happens if this is a single node and its' filesystem becomes read-only?

Reply:
line 451, since there is only one node, it can't add itself into the unresponsive list, there is a condition that checks if a decision-maker and a failed node are the same (which is true in our case), then we can't do anything and just skip the current iteration of FD
```
if (decisionMaker.equals(failedNode.getEndpoint())) {
    log.error("Decision maker and failed node are the same node: {}", decisionMakerAgent);
    return DetectorTask.NOT_COMPLETED;
}
```